engines/base/src/test/java/org/alfresco/transform/base/transform/StreamHandlerTest.java

```diff
@@ -0,0 +1,576 @@
+            transformManager.deleteSourceFile();
+            transformManager.deleteTargetFile();
+
+            assertEquals(EXPECTED, read(outputStream));
+            assertEquals(EXPECTED.length(), outputLength);
+            assertFalse(sourceFileCreatedByTransform.exists());
+        }
+    }
+
+    @Test
+    public void testStartWithSourceFile() throws Exception
+    {
+        File sourceFile = tempFile();
+        write(sourceFile, ORIGINAL);
+        transformManager.setSourceFile(sourceFile);
+
+        try (InputStream inputStream = new BufferedInputStream(new FileInputStream(sourceFile));
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream())
+        {
+            transformManager.setInputStream(inputStream);
+            OutputStream outputStreamLengthRecorder = transformManager.setOutputStream(outputStream);
+
+            write(outputStreamLengthRecorder, read(inputStream)+CHANGE);
+
+            transformManager.copyTargetFileToOutputStream();
+            closeInputStreamWithoutException(inputStream);
+            transformManager.getOutputStream().close();
+            Long outputLength = transformManager.getOutputLength();
+            transformManager.deleteSourceFile();
+            transformManager.deleteTargetFile();
+
+            assertEquals(EXPECTED, read(outputStream));
+            assertEquals(EXPECTED.length(), outputLength);
+            assertFalse(sourceFile.exists());
+        }
+    }
+
+    @Test
+    public void testStartWithSourceFileAndCallCreateSourceFile() throws Exception
+    {
+        File sourceFile = tempFile();
+        write(sourceFile, ORIGINAL);
+        transformManager.setSourceFile(sourceFile);
+
+        try (InputStream inputStream = new BufferedInputStream(new FileInputStream(sourceFile));
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream())
+        {
+            transformManager.setInputStream(inputStream);
+            OutputStream outputStreamLengthRecorder = transformManager.setOutputStream(outputStream);
+
+            File sourceFileCreatedByTransform = transformManager.createSourceFile();
+            assertEquals(sourceFile, sourceFileCreatedByTransform);
+            write(outputStreamLengthRecorder, read(sourceFileCreatedByTransform)+CHANGE);
+
+            transformManager.copyTargetFileToOutputStream();
+            closeInputStreamWithoutException(inputStream);
+            transformManager.getOutputStream().close();
+            Long outputLength = transformManager.getOutputLength();
+            transformManager.deleteSourceFile();
+            transformManager.deleteTargetFile();
+
+            assertEquals(EXPECTED, read(outputStream));
+            assertEquals(EXPECTED.length(), outputLength);
+            assertFalse(sourceFile.exists());
+        }
+    }
+
+    @Test
+    public void testStartWithOutputStream()
+    {
+        // Do nothing. Same as testUsingInputStream() which just uses the outputStream
+    }
```

Comment:
Changed the comment to the following. Hopefully that is better.

        // This method exists so that we have a test for each input or output type. However, it contains no code
        // because it would be identical to the testStartWithInputStream method. Testing without both and input
        // and output would be far more complicated.